### PR TITLE
chore(stream-model-instance): Stricter TS

### PR DESCRIPTION
**Pull request by ukstv**

### CODE REVIEW

File: packages/stream-model-instance/src/model-instance-document.ts [hunk 1]
```diff
@@ -100,13 +101,17 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
...
-  get content(): T {
+  override get content(): T {
     return super.content
   }
 
   get metadata(): ModelInstanceDocumentMetadata {
     const metadata = this.state$.value.metadata
-    return { controller: metadata.controllers[0], model: metadata.model }
+    const controller = metadata.controllers[0]
+    if (!controller) throw new Error(`Controller is not present`)
```
Comment (stbrody) [inline packages/stream-model-instance/src/model-instance-document.ts:111]: that shouldn't be possible, if controller wasn't present the handler should have failed when the stream was loaded

File: packages/stream-model-instance/src/model-instance-document.ts [hunk 2]
```diff
@@ -100,13 +101,17 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
...
+    const model = metadata.model
+    if (!model) throw new Error(`Model is not present`)
```
Comment (stbrody) [inline packages/stream-model-instance/src/model-instance-document.ts:113]: same for model, both of these should be checked by the model-instance-document-handler as part of loading the stream in the first place.  If it gets far enough to have a ModelInstanceDocument instance, this should already be guaranteed to be set

File: packages/stream-model-instance/src/model-instance-document.ts [hunk 3]
```diff
@@ -242,10 +247,12 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
    */
   private _makeRawCommit(newContent: T | null): RawCommit {
     const patch = jsonpatch.compare(this.content, newContent || {})
+    const genesisLogEntry = this.state.log[0]
+    if (!genesisLogEntry) throw new Error(`Genesis log entry is absent`)
```
Comment (stbrody) [inline packages/stream-model-instance/src/model-instance-document.ts:251]: this also should be guaranteed to be set

**Review by stbrody — COMMENTED** [review]
a lot of these changes don't seem like they should be necessary based on the logic of how the code works.  Is the problem that we're not using the type system right so we're getting compiler errors about things that aren't actually issues?

**Comment by ukstv** [comment]
Type system is definitely mis-used. The changes here just highlight pitfalls in reasoning behind the types.

**Comment by stbrody** [comment]
> Type system is definitely mis-used. The changes here just highlight pitfalls in reasoning behind the types.

what would it take to fix the types instead of having to band-aid every time we use them?

**Pull request closed without merging**